### ISSUE TITLE
State tool install failed 

### ISSUE THREAD

**Issue by ferox** [Komodo/KomodoEdit#3929]
### Short Summary
After the installation, when the user finished the wizard's setup there is an error in Notifications
State tool install failed Please contact ActiveState or review your logs

### Steps to Reproduce
First you need to run the script to install.
Then you need to run the komodo bin file
When the user finished the wizard there is an error in Notifications

### Expected results
No error after installation

### Actual results
An error in state

### Platform Information

Komodo IDE, version 12.0.1, build 91869, platform linux-x86_64.
Ubuntu 20.04 LTS
Built on Mon Feb 10 19:37:24 2020.* 

### Additional Information

Error log paste: https://paste.ubuntu.com/p/9DfbMNmstT/

**Comment by th3coop** [comment]
@ferox , is this just an error you're noticing  or is this effecting you being able to use Komodo?

**Comment by ferox** [comment]
No, but it is annoying!!

**Comment by mjross** [comment]
I'm seeing it as well. Screenshot attached.

![Clipboard01](https://user-images.githubusercontent.com/829492/115935166-377dca00-a458-11eb-9ba8-05f2c6eedf85.jpg)

Platform information: Komodo IDE 12.0.1 build 91869 (on Windows Home 10.0.19041.746)

@th3coop: As always, please let me know if you would like me to email to you my logs file or anything else.

**Comment by th3coop** [comment]
@mjross, send your logs.  I'd be interested if you get the same error on Windows.  I've reported the issue to the State Tool team at ActiveState for the issue @ferox  reported.  ie:
```
Warning
───────
err_prepare_completions
```

**Comment by mjross** [comment]
Will do, @th3coop.

**Comment by th3coop** [comment]
@mjross, got your email.  Can you try manually running the install command?
Open a CMD window and run the following command in it:
`powershell -Command "& $([scriptblock]::Create((New-Object Net.WebClient).DownloadString('https://platform.activestate.com/dl/cli/install.ps1')))"`

The error in your logs is different from Ferox's error and I can't reproduce the error you're getting.   Also, what version of Windows are you on @mjross ?

**Comment by mjross** [comment]
I'm now running Microsoft Windows version 10.0.19042.928 (it was updated yesterday).

Running the install command results in:
```
ActiveState collects usage statistics and diagnostic data about failures. The collected data complies with ActiveState Privacy Policy (https://www.activestate.com/company/privacy-policy/) and will be used to identify product enhancements, help fix defects, and prevent abuse.

By running the State Tool installer you consent to the Privacy Policy. This is required for the State Tool to operate while we are still in beta.

╔═══════════════════════╗
║ Installing State Tool ║
╚═══════════════════════╝

Installing to 'C:\Users\Someone\AppData\Roaming\ActiveState\bin'...

Continue? [y/N]
: y
NOTE: C:\Users\Someone\AppData\Roaming\ActiveState\bin will be created

Preparing for installation...

Determining latest version...

Fetching the latest version: 0.25.1-SHAd98b4c4...

Verifying checksums...

Extracting windows-amd64.zip...

Updating state tool:  Downloading latest version of the state tool...

To avoid auto updating run state update --lock (only recommended for production
environments) or set environment variable ACTIVESTATE_CLI_DISABLE_UPDATES=true

Auto Update
───────────
Updated from 0.25.1-SHAd98b4c4 to 0.27.1-SHA26de872.

╔══════════════════════════════╗
║ Exporting Configuration Data ║
╚══════════════════════════════╝
Updating environment...

Adding C:\Users\Someone\AppData\Roaming\ActiveState\bin to user PATH

State Tool successfully installed to: C:\Users\Someone\AppData\Roaming\ActiveState\bin.
Please close your current terminal window and open a CMD prompt in order to start using the 'state.exe' program.  Powershell support is coming soon.
```

By the way, I'm not sure what this 'state' command is used for. So I hope by running this it did not reset or mess up my Komodo IDE settings? I started Komodo, and everything looks fine.

Anyway, when I did start Komodo, it did not display the yellow notification message mentioned above. Let me know if you want me nonetheless to send my new log file.

**Comment by th3coop** [comment]
Thanks for doing that @mjross.  Installing the State Tool won't effect your Komodo install other than add some features that I don't think you'll use given how you use Komodo.  If you'd like to uninstall it now you can run `state clean uninstall` in a CMD window.

So your issue is different and Windows specific. I don't know why it's happening though if we can run the install command manually.  It is somehow related to how Komodo is running the install script, so not a state tool bug like this ticket is.  @mjross  I seem to recall you sending me logs regarding a similar issue but I can't find it.  Do you know which one that is?  If it's not specifically this one we'll need to open a new ticket.

**Comment by mjross** [comment]
Thanks for your reply and the info. As for the related issue, could it be this one: https://github.com/Komodo/KomodoEdit/issues/3276 ? I fear not, because I don't remember sending you any logs regarding that one. But I can't think of any others in which I was discussing some sort of pop-up message.

Regarding my Windows-specific version of this issue affecting only my installation (naturally! :-) , it's not causing me any problem, so let's not bother with any new issue. I just got involved in this particular discussion as I have noticed the same behavior mentioned in the issue title.

By the way, that Ubuntu Pastebin service looks handy and probably a better way of sending log files in the future, because it keeps the discussion within the particular thread and doesn't require the use of a separate email message.

**Comment by mjross** [comment]
Just now I ran ```C:\Users\Someone\AppData\Roaming\ActiveState\bin\state clean uninstall```
and got...
```
╔══════════════╗
║ Uninstalling ║
╚══════════════╝

Please Confirm
You are about to remove the State Tool, installed language runtimes, and all configuration information. Continue? (y/N)
> y

Something Went Wrong
────────────────────
 x Could not remove State Tool cache directory

Need More Help?
───────────────
 • Run → `state uninstall --help` for general help
 • Visit the Forum → https://community.activestate.com/c/state-tool/
```

**Comment by mjross** [comment]
@th3coop, I'm now realizing to my horror that I shouldn't have run ```state clean uninstall```. After restarting Komodo, all my macros and settings are gone. :-(  Perhaps I misunderstood your instructions. Anyway, I'm trying to get back to the previous environment by restoring from backup files.

**Comment by th3coop** [comment]
@mjross uninstalling state tool will not effect Komodo.  Something else happened in your env if your Komodo env is borked.  The state tool doesn't know anything about Komodo and it's settings, configs, etc aren't saved anywhere near Komodo's.

Did you check to see if your profile folder is intact?

**Comment by mjross** [comment]
Would my profile consist of all the files in `C:\Users\Someone\AppData\Local\ActiveState\KomodoIDE\12.0` ? That's where I find the `prefs.xml` file and the `schemes` and `tools` directories. Everything in that directory was reset to the default files of a fresh installation, the moment that I closed Komodo. I'm sure that it's related to the installation and/or uninstallation of the state tool, because all of my preferences and macros were present late yesterday morning, when I did a daily backup, and I did not work on my laptop after that time. (It was those backup copies that I used to restore everything.)

In the uninstall instructions, I see the phrase "You are about to remove […] all configuration information." Are you 100% sure that that command couldn't have changed my non-state configuration?

**Comment by th3coop** [comment]
@mjross correct, that is where your profile is. Sorry for not providing details, I figured you'd know where profiles are!  I'm following up with the State Tool team about this.  That's scary if that's in fact what happened.

**Comment by mjross** [comment]
No problem regarding the identity of the Komodo profile. I was just uncertain because the phrase "profile folder" made me think of a Windows folder named `profile`. But you meant it in the descriptive sense.

Thank you in advance for following up on the State Tool. I can attest that it certainly was scary to me! :-)
